src/TidePublicationOperation.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace Drupal\tide_publication;
+
+use Drupal\Core\Config\Config;
+use Drupal\field\Entity\FieldConfig;
+use Drupal\taxonomy\Entity\Vocabulary;
+
+/**
+ * Tide publication modules operations.
+ */
+class TidePublicationOperation {
+
+  /**
+   * The bundle name.
+   */
+  const BUNDLE = 'publication';
+
+  /**
+   * The entity reference field type.
+   */
+  const ENTITY_REFERENCE_FIELD_TYPE = 'entity_reference';
+
+  /**
+   * The publication authors field name.
+   */
+  const PUBLICATION_AUTHORS_FIELD_NAME = 'field_publication_authors';
+
+  /**
+   * The publication authors field description.
+   */
+  const PUBLICATION_AUTHORS_FIELD_DESCRIPTION = 'Start typing to choose one publication author.';
+
+  /**
+   * The publication authors field storage.
+   */
+  const PUBLICATION_AUTHORS_FIELD_STORAGE = 'field.storage.node.field_publication_authors';
+
+  /**
+   * Update field_publication_authors to use new vocabulary.
+   *
+   * @param \Drupal\Core\Config\Config $config
+   *   The config object.
+   * @param string $vocabulary
+   *   The new vocabulary.
+   */
+  public function updateFieldPublicationAuthorsVocabulary(Config $config, string $vocabulary) {
+    // Creates vocabulary if doesn't exist.
+    $vocabularies = Vocabulary::loadMultiple();
+    if (!isset($vocabularies[$vocabulary])) {
+      $this->createVocabulary($vocabulary);
+    }
+
+    $field = FieldConfig::loadByName('node', self::BUNDLE, self::PUBLICATION_AUTHORS_FIELD_NAME);
+    if ($field->get('field_type') === self::ENTITY_REFERENCE_FIELD_TYPE) {
+      $new_field = $field->toArray();
+      $new_field['field_type'] = self::ENTITY_REFERENCE_FIELD_TYPE;
+      $new_field['description'] = self::PUBLICATION_AUTHORS_FIELD_DESCRIPTION;
+      $new_field['dependencies'] = [
+        'config' => [
+          'field.storage.' . $config->get('id'),
+          'node.type.' . self::BUNDLE,
+          'taxonomy.vocabulary.' . $vocabulary,
+        ],
+      ];
+      $new_field['settings'] = [
+        'handler_settings' => [
+          'target_bundles' => [
+            'department' => $vocabulary,
+          ],
+        ],
+      ];
+      $new_field = FieldConfig::create($new_field);
+      $new_field->original = $field;
+      $new_field->enforceIsNew(FALSE);
+      $new_field->save();
+    }
+  }
+
+  /**
+   * Creates a new vocabulary.
+   *
+   * @param string $vocabulary
+   *   The new vocabulary.
+   */
+  public function createVocabulary(string $vocabulary) {
```

Comment:
This helper function we can keep, cause this can be used for creating any other vocabulary.

Reply:
This is a good example of a code reuse pattern, good one @yeniatencio